### PR TITLE
virsh_cpu_xml: add cases for cpu

### PR DESCRIPTION
**Pull request by dzhengfy**
RHEL-150995: Test "virsh {hypervisor-}cpu-{baseline,compare}" cmds with XML file with XML declaration
RHEL-139122: Test 'virsh hypervisor-cpu-baseline' cmd with <cpu>/domcapabilities/capabilities XML
RHEL-139119: Test 'virsh cpu-baseline' cmd with <cpu> definition / capabilities XML

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/3189

**Comment by dzhengfy** [comment]
Run with libvirt-7.6.0-1.module+el8.5.0+12097+2c77910b.x86_64 on ci-osp vm.
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.cpu_xml
JOB ID     : a26b4e4acc27c1b5dab1befdd5e5b05fd4499fd2
JOB LOG    : /root/avocado/job-results/job-2021-08-23T00.00-a26b4e4/job.log
 (01/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_compare.xml_declaration: PASS (43.55 s)
 (02/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.xml_declaration: PASS (44.94 s)
 (03/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.cpu_xml: PASS (46.07 s)
 (04/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.domcap_xml: PASS (44.73 s)
 (05/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.cap_xml: PASS (44.96 s)
 (06/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_compare.xml_declaration: PASS (45.74 s)
 (07/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.xml_declaration: PASS (43.48 s)
 (08/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.cap_xml: PASS (44.63 s)
 (09/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.cap_cpu_xml: PASS (44.76 s)
 (10/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.negative.cpu_baseline.cap_xml_mix: PASS (43.68 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 447.48 s

```

**Comment by dzhengfy** [comment]
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.cpu_xml
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : e6a54fabecc23144566d71c3289bb81683e3ba3a
JOB LOG    : /root/avocado/job-results/job-2021-09-16T20.41-e6a54fa/job.log
 (01/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.xml_declaration: PASS (253.68 s)
 (02/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.cpu_xml: PASS (43.73 s)
 (03/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.domcap_xml: PASS (42.16 s)
 (04/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.hypervisor_cpu_baseline.cap_xml: PASS (45.53 s)
 (05/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.xml_declaration: PASS (50.12 s)
 (06/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.cap_xml: PASS (60.36 s)
 (07/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.positive.cpu_baseline.cap_cpu_xml: PASS (57.69 s)
 (08/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.negative.cpu_baseline.cap_xml_mix: PASS (59.41 s)
 (09/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.negative.hypervisor_cpu_compare.xml_declaration: PASS (56.68 s)
 (10/10) type_specific.io-github-autotest-libvirt.virsh.cpu_xml.negative.cpu_compare.xml_declaration: PASS (54.80 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 725.69 s

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                - cpu_compare:
+                    only xml_declaration
+                    virsh_function = 'virsh.cpu_compare'
+                    err_msg = 'CPU described in.*is identical to host CPU|CPU described in.*is incompatible with host CPU'
+                - cpu_baseline:
+                    virsh_function = 'virsh.cpu_baseline'
+        - negative:
+            variants:
+                - cap_xml_mix:
+                    only
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg]: Update

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                - cap_xml:
+                    file_path = 'libvirt/tests/deps/capabilities.xml'
+                - cap_cpu_xml:
+                    only cpu_baseline
+                    file_path = 'libvirt/tests/deps/capability_cpu.xml'
+            variants:
+                - hypervisor_cpu_compare:
+                    only xml_declaration
+                    virsh_function = 'virsh.hypervisor_cpu_compare'
+                    err_msg = 'CPU described.*is incompatible with the CPU provided by hypervisor on the host'
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg]: Good catch. The exit code is 1, but error message is printed into stdout. I move these cases into negative part.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            variants:
+                - xml_declaration:
+                    file_xml_declaration = "<?xml version='1.0' encoding='UTF-8'?>"
+                    file_path = 'libvirt/tests/deps/capabilities.xml'
+                - cpu_xml:
+                    no cpu_baseline
+                    file_path = 'libvirt/tests/deps/cpu.xml'
+                - domcap_xml:
+                    no cpu_baseline
+                    file_path = 'libvirt/tests/deps/domcapabilities.xml'
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg:14]: No

[the remark this answers is not in the document]
Comment (Yingshun) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_cpu_xml.cfg:14]: Ok, got it. Thanks for your explanation!

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by chloerh — APPROVED** [review]

**Pull request merged**